src/routes/safe/components/CurrencyDropdown/index.tsx

```diff
@@ -43,9 +46,10 @@ export const CurrencyDropdown = (): React.ReactElement | null => {
     setAnchorEl(null)
   }
 
-  const onCurrentCurrencyChangedHandler = (newCurrencySelectedName: string) => {
-    dispatch(setSelectedCurrency({ selectedCurrency: newCurrencySelectedName }))
+  const onCurrentCurrencyChangedHandler = async (newCurrencySelectedName: string): Promise<void> => {
     handleClose()
+    await dispatch(fetchSafeTokens(address, newCurrencySelectedName))
```

Comment:
This is basically the fix. The other changes are just some minor things I noticed.